pyccel/codegen/wrapper/fortran_to_c_wrapper.py

```diff
@@ -315,59 +263,149 @@ def _wrap_Interface(self, expr):
+        self.scope.insert_symbol(name)
+        collisionless_name = self.scope.get_expected_name(var.name)
+        f_arg = var.clone(collisionless_name, new_class = Variable, is_argument = False,
+                is_optional = False, memory_handling='alias')
+        new_var = Variable(BindCPointer(), self.scope.get_new_name(f'bound_{name}'),
+                            is_argument = True, is_optional = False, memory_handling='alias')
+        body = [C_F_Pointer(new_var, f_arg)]
+        self.scope.insert_variable(f_arg)
+        return {'c_arg': BindCVariable(new_var, var), 'f_arg': f_arg, 'body': body}
+
+    def _extract_NumpyNDArrayType_FunctionDefArgument(self, var, func):
+        name = var.name
+        scope = self.scope
+        scope.insert_symbol(name)
+        collisionless_name = scope.get_expected_name(var.name)
+        rank = var.rank
+        order = var.order
+        bind_var = Variable(BindCPointer(), scope.get_new_name(f'bound_{name}'),
+                            is_argument = True, is_optional = False, memory_handling='alias')
+        arg_var = var.clone(collisionless_name, is_argument = False, is_optional = False,
+                            memory_handling = 'alias', allows_negative_indexes=False,
+                            new_class = Variable)
+        scope.insert_variable(arg_var)
+        scope.insert_variable(bind_var)
+
+        shape   = [scope.get_temporary_variable(PythonNativeInt(), name=f'{name}_shape_{i+1}', is_argument = True)
+                   for i in range(rank)]
+        stride  = [scope.get_temporary_variable(PythonNativeInt(), name=f'{name}_stride_{i+1}', is_argument = True)
+                   for i in range(rank)]
+
+        orig_size = [PyccelMul(sh,st) for sh,st in zip(shape, stride)]
+        body = [C_F_Pointer(bind_var, arg_var, orig_size[::-1] if order == 'C' else orig_size)]
+
+        c_arg_var = Variable(BindCArrayType(rank, has_strides = True),
+                        scope.get_new_name(), is_argument = True,
+                        shape = (LiteralInteger(rank*2+1),))
+
+        scope.insert_symbolic_alias(IndexedElement(c_arg_var, LiteralInteger(0)), bind_var)
+        for i,s in enumerate(shape):
+            scope.insert_symbolic_alias(IndexedElement(c_arg_var, LiteralInteger(i+1)), s)
+        for i,s in enumerate(stride):
+            scope.insert_symbolic_alias(IndexedElement(c_arg_var, LiteralInteger(i+rank+1)), s)
+
+        start = LiteralInteger(1) # C_F_Pointer leads to default Fortran lbound
+        stop = None
+        indexes = [Slice(start, stop, step) for step in stride]
+
+        f_arg = IndexedElement(arg_var, *indexes)
+
+        return {'c_arg': BindCVariable(c_arg_var, var), 'f_arg': f_arg, 'body': body}
+
+    def _extract_HomogeneousTupleType_FunctionDefArgument(self, var, func):
+        name = var.name
+        scope = self.scope
+        scope.insert_symbol(name)
+        collisionless_name = scope.get_expected_name(var.name)
```

Comment:
`var.name` can be replaced with `name`

Reply:
Fixed [f9365af](https://github.com/pyccel/pyccel/pull/2232/commits/f9365af03c0482c7585c3d4288b8df0b6307a067)